exalearn-api/ExaLearn.Bl/Services/AudioFileService.cs

```diff
@@ -10,29 +10,32 @@
 using ExaLearn.Dal.Interfaces;
 using System.ComponentModel.DataAnnotations;
 using System;
+using AutoMapper;
+using ExaLearn.Bl.DTO;
 
 namespace ExaLearn.Bl.Services
 {
-    public class FileService : IFileService
+    public class AudioFileService : IAudioFileService
     {
-        private readonly IGenericRepository<AudioFile> _filesRepository;
+        private readonly IAudioFileRepository _audioFileRepository;
         private readonly IConfiguration _configuration; //it should be in startup.. but for now i do it like this
+        private readonly IMapper _mapper;
 
-        public FileService(IGenericRepository<AudioFile> filesRepository, IConfiguration configuration)
+        public AudioFileService(IAudioFileRepository audioFileRepository, IConfiguration configuration, IMapper mapper)
         {
-            _filesRepository = filesRepository;
+            _audioFileRepository = audioFileRepository;
             _configuration = configuration;
+            _mapper = mapper;
         }
 
-        public async Task<AudioFile> AddAsync(IFormFile file)
+        public async Task<AudioFileDTO> AddAsync(IFormFile file)
         {
             if (file == null && file.Length <= 0)
                 throw new ValidationException("File not found!");
 
             var fileType = Path.GetExtension(file.FileName);
             var isNessesaryFormat = AudioExtensions.AvailableAudutionExtensions.Contains(fileType.ToLower());
 
-
             if (!isNessesaryFormat)
             {
                 var availableFormats = string.Join(", ", AudioExtensions.AvailableAudutionExtensions);
```

Comment:
in the  47 line - think about await using 
look at https://stackoverflow.com/questions/58610350/what-is-the-difference-between-using-and-await-using-and-how-can-i-decide-which